File: mkfuncs.pl
#! /usr/bin/env perl
use strict;
use Getopt::Std;

my $state = 0;
my $def;
my $params;

my %opt;
die "usage: mkfuncs.pl [-p]\n" if not getopts('p', \%opt);
my $passthru = $opt{p};
while (<>) {
	my $match = $passthru ? /^\t(public|static)\s+([^;]+)$/ : /^\t(public)\s+([^;]+)$/;
	if ($match) {
		$def = "$1 $2";
		$state = 1;
		$params = 0;
	} elsif ($state == 1 and /(\w+)\s*\(/) {
		$def .= " $1";
		$def .= " LESSPARAMS (" if not $passthru;
		$def .= "(";
		$state = 2;
	} elsif ($state == 2) {
		if (/^{/) {
			if (not $params) {
				$def .= ($passthru ? 'void' : 'void');
			}
			print $def;
			print ")" if not $passthru;
			print ")\n{\n";
			$state = 0;
		} elsif (/^\s*([^;]*)/) {
			$def .= ', ' if substr($def,-1) ne '(';
			$def .= $1;
			$params = 1;
		}
	} else {
		print if $passthru;
	}
}

File: mkfuncs.py
#!/usr/bin/env python

import fileinput
import re

definition = ''
state = 0
params = 0

for line in fileinput.input():
        if test := re.search(r'^\tpublic\s+(.*)', line):
            definition = 'public ' + test.group(1)
            state = 1
            params = 0
        elif (state == 1) and (test := re.search(r'(\w+)\s*\((.*)\)', line)):
            definition += '{0} LESSPARAMS (({1}))'.format(test.group(1), test.group(2))
            state = 2
        elif state == 2:
            print(f'{definition};')
            state = 0
